tests/testthat/test-trace.R

```diff
@@ -659,3 +659,18 @@ test_that("can bind backtraces", {
     c(0:3, c(4L, 4L))
   )
 })
+
+test_that("backtraces don't contain inlined objects (#1069, r-lib/testthat#1223)", {
+  # !! deparsing in older R
+  skip_if_not_installed("base", "3.5.0")
+
+  local_options(
+    rlang_trace_format_srcrefs = FALSE
+  )
+  e <- environment()
+  f <- function(...) g()
+  g <- function() h()
+  h <- function() trace_back(e)
+  trace <- inject(f(!!list()))
```

Comment:
Could also use `do.call()` here, since that's the most common way that users will create inlined objects.

Reply:
Maybe something like along these lines:

``` r
e <- environment()
f <- function() do.call("g", list(1:1e6 + 1))
g <- function(...) h()
h <- function() rlang::trace_back(e)
lobstr::obj_size(f())
#> 8,014,888 B
```

<sup>Created on 2021-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

(But using something other than `obj_size()`)

Reply:
Maybe 

```R
trace <- f()
trace$calls[[3]]
```

?

Reply:
Good idea, I've added a `do.call()` and size check in the test.